src/style/fog.js

```diff
@@ -29,16 +34,77 @@ const TRANSITION_SUFFIX = '-transition';
 export const FOG_PITCH_START = 55;
 export const FOG_PITCH_END = 65;
 
+export class FogSampler {
```

Comment:
In the symbols branch I implemented this as a cheap-to-clone snapshot of the `Fog` style object. This is because `Placement` works with and snapshot of `Transform` and `Fog` since it has to run over multiple frames.

This seems to take properties object by reference which makes it jankier to clone.

Reply:
Are you suggesting to remove the reference to properties?